### PR TITLE
Add support for openSUSE platform

### PR DESCRIPTION
**Pull request by axl89**
### Description

- Adds support for openSUSE platform

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -16,3 +16,4 @@
 supports 'redhat'
 supports 'oracle'
 supports 'amazon'
+supports 'opensuseleap'
```
Comment (ramereth) [inline metadata.rb:19]: You might as well add `suse` here too.

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/install.rb [hunk 2]
```diff
@@ -3,7 +3,13 @@
...
   include_recipe 'yum-epel' if platform_family?('rhel', 'amazon')
 
-  package %w(haveged gnupg2)
+  gpg2_package_name = if platform?('opensuseleap')
+                        'gpg2'
+                      else
+                        'gnupg2'
+                      end
+
+  package %W(haveged #{gpg2_package_name})
```
Comment (ramereth) [inline resources/install.rb:12]: Can you please move this to `libaries/helpers.rb` as a `gpg2_packages` method?

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please also add a Changelog entry under `## Unreleased`?

File: resources/install.rb (hunk 2 above)
Comment (axl89) [reply resources/install.rb]: Done in 3d06ba3c64ccb44c2295ffba929ce41db93b7910

File: metadata.rb (hunk 1 above)
Comment (axl89) [reply metadata.rb]: Done in e484c6d9dae478983ee498a5b4b78387f5e39caa

File: resources/install.rb (hunk 2 above)
Comment (axl89) [reply resources/install.rb]: I've refactored it a bit in ab2a708606e18ec917b8d5131239537456f3eee1. I think this is what you meant in the first place :sweat_smile:

File: libraries/helpers.rb [hunk 3]
```diff
@@ -29,5 +29,15 @@ def gpg_opts(new_resource)
     def gpg_cmd
       "gpg2 --homedir #{new_resource.home_dir} "
     end
+
+    def gpg2_packages
+      packages = %w(haveged)
+      if platform?('opensuseleap', 'suse')
```
Comment (ramereth) [inline libraries/helpers.rb:35]:
```suggestion
      if platform_family?('suse')
```

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [1.3.0](https://github.com/sous-chefs/gpg/releases/tag/1.3.0)

**Pull request merged**